src/Microsoft.Framework.DesignTimeHost/DesignTimeAssemblyLoadContextFactory.cs

```diff
@@ -62,9 +57,12 @@ private IAssemblyLoadContextFactory GetRuntimeFactory(IServiceProvider servicePr
 
                 // Add a cache dependency on restore complete to reevaluate dependencies
                 ctx.Monitor(_namedDependencyProvider.GetNamedDependency(_project.Name + "_Dependencies"));
-
-                return new AssemblyLoadContextFactory(applicationHostContext.ServiceProvider);
+                
+                return applicationHostContext;
             });
+            
+            var factory = new AssemblyLoadContextFactory(appHostContext.ServiceProvider);
```

Comment:
Caching didn't break it, calling Create with the wrong service provider broke it.

This was the old code:
```
return GetRuntimeFactory(serviceProvider).Create(serviceProvider);
```

The new code calls create with the correct service provider.